base/deprecated.jl

```diff
@@ -1272,6 +1272,11 @@ for f in (:airyai, :airyaiprime, :airybi, :airybiprime, :airyaix, :airyaiprimex,
     end
 end
 
+@deprecate_binding LinearIndexing IndexStyle
```

Comment:
This macro will export the old bindings.  The `@deprecate` macro has an optional third argument to disable this behavior — perhaps you could add that to `@deprecate_binding`, too.